repose-aggregator/components/filters/cors-filter/src/main/scala/org/openrepose/filters/cors/CorsFilter.scala

```diff
@@ -178,23 +204,71 @@ class CorsFilter @Inject()(configurationService: ConfigurationService)
         Nil
     })
   }
+
+  def getHostUri(request: HttpServletRequestWrapper): URI = {
+    val maybeForwardedHost = request.getSplittableHeaderScala(CommonHttpHeader.X_FORWARDED_HOST).headOption
+    val maybeForwardedHostUri = Try(maybeForwardedHost
+      .map(forwardedHost => new URIBuilder(s"${request.getScheme}://$forwardedHost"))
+      .map(uri => uri.setPort(normalizePort(uri.getPort, uri.getScheme)).setHost(normalizeUriHost(uri.getHost)).build()))
+    lazy val hostUri = new URIBuilder().setScheme(request.getScheme).setHost(normalizeHost(request.getServerName))
```

Comment:
Is `request.getServerName`guaranteed to be in-sync with the `Host` header? I would hope so, but I'm not certain.

Actually, I know that isn't the case in our wrappers. Server name can be set independently of the `Host` header and vice-versa. We should probably fix that.